controller-api/src/main/java/com/yahoo/vespa/hosted/controller/api/integration/noderepository/NodeRepositoryNode.java

```diff
@@ -357,8 +358,9 @@ public void setWantedFirmwareCheck(Long wantedFirmwareCheck) {
         this.wantedFirmwareCheck = wantedFirmwareCheck;
     }
 
+    @JsonIgnore
     public Map<String, JsonNode> getReports() {
```

Comment:
If NodeRepositoryNode will be used to serialize to JSON, I suggest adding a method

```
@JsonGetter("reports")
public Map<...> getReportsOrNull() { return reports; }
```

that will be used by jackson to serialize, while our code will typically refer to the `getReports()` to avoid having to deal with `null`.

Reply:
This would be in addition to the changes you've done in this PR so far.

Reply:
Not sure that I get what you mean. Should I revert getReports to its original state?

Reply:
No, my proposed changes are in addition to the ones you have already done in this PR.

Reply:
nice, the getter really saved me the null handling 👍